src/payload.iced

```diff
@@ -1,45 +1,51 @@
-{prng} = require('crypto')
-{createHash} = require('crypto')
-{createHmac} = require('crypto')
-msgpack = require('purepack')
-crypto = require('keybase-nacl')
-nonce = require('./nonce.iced')
-util = require('./util.iced')
+crypto = require('crypto')
+{make_esc} = require('iced-error')
+nonce = require('./nonce')
+util = require('./util')
 
 # HMAC-SHA512-256(step1 hash, mac_key)
-compute_authenticator = (hash, key) ->
-  hmac = createHmac('sha512', key)
-  hmac.update(hash)
-  return hmac.digest()[0...32]
+compute_authenticator = ({hash, key}, cb) ->
+  try
+    hmac = crypto.createHmac('sha512', key)
+    hmac.update(hash)
+    authenticator = hmac.digest()[0...32]
+  catch
```

Comment:
I don't think we want to - we want to catch **any** errors thrown by both `hmac` methods, and freak out if they fail.